c/zis/services/auth.c

```diff
@@ -35,32 +35,68 @@ static int handleVerifyPassword(AuthServiceParmList *parmList,
   int safRC = 0, racfRC = 0, racfRsn = 0;
   int deleteSAFRC = 0, deleteRACFRC = 0, deleteRACFRsn = 0;
   int rc = RC_ZIS_AUTHSRV_OK;
-  IDTA *idta = NULL;
-  int options = VERIFY_CREATE;
-
-  if (parmList->options & ZIS_AUTH_SERVICE_PARMLIST_OPTION_GENERATE_IDT) {
-    idta = (IDTA *) safeMalloc31(sizeof(IDTA), "Idta structure");
-    memset(idta, 0, sizeof(IDTA));
-    memcpy(idta->id, "IDTA", 4);
-    idta->version = IDTA_VERSION_0001;
-    idta->length = sizeof(IDTA);
-    idta->idtType = IDTA_JWT_IDT_Type;
-    idta->idtBufferPtr = parmList->safIdt;
-    idta->idtBufferLen = sizeof(parmList->safIdt);
-    idta->idtLen = parmList->safIdtLen;
-    idta->idtPropIn = IDTA_End_User_IDT;
-    options |= VERIFY_GENERATE_IDT;
-  }
 
   CMS_DEBUG(globalArea, "handleVerifyPassword(): username = %s, password = %s\n",
       parmList->userIDNullTerm, "******");
-  if (idta == NULL) {
-    safRC = safVerify(VERIFY_CREATE, parmList->userIDNullTerm,
+
```

Comment:
Since you've reverted changes in this function, it no longer support the IDT option, doesn't it? I think the right thing to do would be checking
```c
if (parmList->options & ZIS_AUTH_SERVICE_PARMLIST_OPTION_RESERVED) {
  return SOME_BAD_RC;
```
That way you'll make sure that a previous version of ZSS which relied on this would fail explicitly.

Reply:
added check and return bad version rc

Reply:
You need to check that **before** you do any SAF calls.

Reply:
moved higher